oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/DocumentPropertyStateTest.java

```diff
@@ -81,4 +100,179 @@ public void multiValuedBinarySize() throws Exception {
+
+        reads.clear();
+        assertEquals(10050, p.size(0));
+        // must not read the string via streams
+        assertEquals(0, reads.size());
+    }
+
+    @Test
+    public void compressValueThrowsException() throws IOException, NoSuchFieldException, IllegalAccessException {
+        DocumentNodeStore mockDocumentStore = mock(DocumentNodeStore.class);
+        Compression mockCompression = mock(Compression.class);
+        when(mockCompression.getOutputStream(any(OutputStream.class))).thenThrow(new IOException("Compression failed"));
+
+        Field compressionThreshold = DocumentPropertyState.class.getDeclaredField("DEFAULT_COMPRESSION_THRESHOLD");
+        compressionThreshold.setAccessible(true);
+        Field modifiersField = Field.class.getDeclaredField("modifiers");
+        modifiersField.setAccessible(true);
+        modifiersField.setInt(compressionThreshold, compressionThreshold.getModifiers() & ~Modifier.FINAL);
+
+        compressionThreshold.set(null, DEFAULT_COMPRESSION_THRESHOLD);
+
+        DocumentPropertyState documentPropertyState = new DocumentPropertyState(mockDocumentStore, "p", "\"" + STRING_HUGEVALUE + "\"", mockCompression);
+
+        assertEquals(documentPropertyState.getValue(Type.STRING), STRING_HUGEVALUE);
+
+        verify(mockCompression, times(1)).getOutputStream(any(OutputStream.class));
+
+        compressionThreshold.set(null, -1);
+
+    }
+
+    @Test
+    public void uncompressValueThrowsException() throws IOException, NoSuchFieldException, IllegalAccessException {
+
+        DocumentNodeStore mockDocumentStore = mock(DocumentNodeStore.class);
+        Compression mockCompression = mock(Compression.class);
+        OutputStream mockOutputStream= mock(OutputStream.class);
+        when(mockCompression.getOutputStream(any(OutputStream.class))).thenReturn(mockOutputStream);
+        when(mockCompression.getInputStream(any(InputStream.class))).thenThrow(new IOException("Compression failed"));
+
+        Field compressionThreshold = DocumentPropertyState.class.getDeclaredField("DEFAULT_COMPRESSION_THRESHOLD");
+        compressionThreshold.setAccessible(true);
+        Field modifiersField = Field.class.getDeclaredField("modifiers");
+        modifiersField.setAccessible(true);
+        modifiersField.setInt(compressionThreshold, compressionThreshold.getModifiers() & ~Modifier.FINAL);
+
+        compressionThreshold.set(null, DEFAULT_COMPRESSION_THRESHOLD);
+
+        DocumentPropertyState documentPropertyState = new DocumentPropertyState(mockDocumentStore, "p", STRING_HUGEVALUE, mockCompression);
+
+        assertEquals(documentPropertyState.getValue(Type.STRING), "{}");
+
+        verify(mockCompression, times(1)).getInputStream(any(InputStream.class));
+
+        compressionThreshold.set(null, -1);
+    }
+
+    @Test
+    public void adefaultValueSetToMinusOne() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException, NoSuchFieldException {
```

Comment:
was wondering if this method should also test that `-1` indeed "does nothing" - i.e. it wouldn't call length()/getBytes()/compress() - not sure if we need to cover all these cases, but adding something that would have failed in the previous iteration might be useful. PS: is there reason the method starts with adefault?

Reply:
The method test that for -1, the compression is not made.

Reply:
I'd suggest to avoid relying on test order in this case - and ensure the test class doesn't leave `DocumentPropertyState` in a changed state. Otherwise other test classes will be affected unintentionally.

Reply:
Done